lib/rubygems/ext/builder.rb

```diff
@@ -17,7 +17,7 @@ def self.class_name
     $1.downcase
   end
 
-  def self.make(dest_path, results, make_dir = Dir.pwd, sitedir = nil)
+  def self.make(dest_path, results, make_dir = Dir.pwd, targets = ["clean", "", "install"])
```

Comment:
I don't know what `sitedir` mean... is it being used?

Anyway, I would not modify the signature of the method this way, may be adding a keyword argument to make sure to prevent misunderstandings.

Reply:
The `sitedir` parameter was only introduced very recently at https://github.com/rubygems/rubygems/commit/dea41fa2dcce8a6c11c7c92c73db7a66a7d7ed51. I'm aware our extension building code is quite complicated, but it used to be worse 😅.

The above change was released with RubyGems 3.3.17, quite recently, so I don't think removing changing it with another positional parameter is going to break anything.

That said, I want to get to a green base line first, then figure out the details.

Reply:
This refactorings broke Windows 2.6 and Windows 2.7 for some reason. I really don't want to investigate that so I removed those simplifications and will be opening a separate PR that can be merged by the end of next year when we will drop Ruby 2.6 and Ruby 2.7 support.